go/caddy/pfsso/pfsso.go

```diff
@@ -0,0 +1,116 @@
+package pfsso
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/fingerbank/processor/log"
+	"github.com/fingerbank/processor/statsd"
+	"github.com/inverse-inc/packetfence/go/caddy/caddy"
+	"github.com/inverse-inc/packetfence/go/caddy/caddy/caddyhttp/httpserver"
+	"github.com/inverse-inc/packetfence/go/firewallsso"
+	"github.com/inverse-inc/packetfence/go/pfconfigdriver"
+	"github.com/julienschmidt/httprouter"
+	"net/http"
+	"runtime/debug"
+	"strconv"
+)
+
+func init() {
+	caddy.RegisterPlugin("pfsso", caddy.Plugin{
+		ServerType: "http",
+		Action:     setup,
+	})
+}
+
+func (h PfssoHandler) handleStart(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	ctx := r.Context()
+	defer statsd.NewStatsDTiming(ctx).Send("PfssoHandler.handleStart")
+
+	var info map[string]string
+	json.NewDecoder(r.Body).Decode(&info)
+
+	for _, firewall := range h.firewalls {
+		// Creating a local shallow copy to send to the go-routine
+		firewall := firewall
+		go func() {
+			timeout, err := strconv.ParseInt(info["timeout"], 10, 32)
+			if err != nil {
+				log.LoggerWContext(ctx).Warn(fmt.Sprintf("Can't parse timeout '%s' into an int (%s). Will not specify timeout for request.", info["timeout"], err))
+			}
+			firewallsso.ExecuteStart(ctx, firewall, info, int(timeout))
+		}()
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+}
+
+func (h PfssoHandler) handleStop(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	ctx := r.Context()
+	defer statsd.NewStatsDTiming(ctx).Send("PfssoHandler.handleStart")
```

Comment:
Should handleStop be sending a statsd time for handleStart?
Am I missing something obvious?